test/integration/test/webpush_test.js

```diff
@@ -0,0 +1,165 @@
+
+      return chakram.post(loginURL,null,{'headers' : header})
+      .then(function(resp){
+       header.Authorization = 'Bearer ' + resp.body.session_token;
+       chakram.setRequestDefaults({'headers': header});
+     });
+    });
+
+    it('Check webpush service is registered',function(){  
+      var pushService = false;
+
+      return chakram.get(serviceListURL)
+      .then(function(listResponse) {
+        for (var entry of listResponse.body) {
+          if (entry.adapter === 'webpush@link.mozilla.org') {
+              pushService = true;                
+           }
+        }
+        expect(listResponse).to.have.status(200);
+        expect(pushService).equals(true);
+      });
+    });
+
+    it('Add push subscription',function(){     
+      var pushURI = Prepper.webPush_server.getEndpointURI();
+      var pushkey = Prepper.webPush_server.getPublicKey();
+      var setter = 'setter:subscribe.webpush@link.mozilla.org';
+      var payload = {
+        'select': {'id':setter}, 
+        'value': 
+          {'Json': 
+            {'subscriptions':[{
+              'push_uri':pushURI,
+              'public_key':pushkey}]}}};
+
+        return chakram.put(setterURL,payload)
+        .then(function(cmdResp){
+         expect(cmdResp).to.have.status(200);
+         expect(cmdResp.body[setter]).equals(null);
+       });
+    });
+    
+    it('Verify push subscription',function(){   
+      var getter =  'getter:subscription.webpush@link.mozilla.org';
+      var payload = {'id': getter};    
+      
+      return chakram.put(getterURL,payload)
+      .then(function(cmdResp){
+        expect(cmdResp).to.have.status(200);
+        expect(cmdResp.body[getter].Json.subscriptions[0].public_key).equals(
+        Prepper.webPush_server.getPublicKey());
+        expect(cmdResp.body[getter].Json.subscriptions[0].push_uri).equals(
+        Prepper.webPush_server.getEndpointURI());
+     });
+    });
+
+    it('Set resources to receive notification',function(){        
+      var setter = 'setter:resource.webpush@link.mozilla.org';
+      var payload = {
+        'select': {
+          'id': setter}, 
+        'value': {
+          'Json': {
+            'resources':['livingroom', 'washroom']}}};
+
+        return chakram.put(setterURL,payload)
+        .then(function(cmdResp){
+         expect(cmdResp).to.have.status(200);
+         expect(cmdResp.body[setter]).equals(null);
+       });
+    });
+ 
+    it('Verify notification resources',function(){        
+      var getter =  'getter:resource.webpush@link.mozilla.org';
+      var payload = {'id': getter};    
+      
+      return chakram.put(getterURL,payload)
+      .then(function(cmdResp){
+        expect(cmdResp).to.have.status(200);
+        expect(cmdResp.body[getter].Json.resources[0]).equals('livingroom');
+        expect(cmdResp.body[getter].Json.resources[1]).equals('washroom');
+     });
+    });
+
+    it('Trigger notification',function(){
```

Comment:
I'm not sure I understood the test correctly but maybe we could make the expectation clearer. How about something like `accepts notification trigger requests`?